### DISCUSSION TITLE
Dynamically Set Access Token on Find, Get, Create, Remove Requests | Socketio | Rest | Feathers Client

### DISCUSSION THREAD

**Discussion by palmtown** [feathersjs/feathers#2541] [Q&A] [answered]
Hello community,

I was hoping to lobby your assistance.  I would like to dynamically set the access token while using the feathers socketio or rest client for get, find, remove, create and update requests.  After initial login, I would like the ability to change the auth token dynamically.  An example would be below.  Any help would be greatly appreciated.

```
app.service("messages").find({
                    query: query,
                     accessToken: accessToken
                    strategy: "jwt"                    
                }).then((results) => {
                   // do something with response
                }).catch(e => {
                   // do something with error
                });
```

**Comment by daffl** [comment, marked as answer]
Yes, you can do this dynamically by setting `params.authentication` before the `authenticate('jwt')` hook. In your case you would pull the access token from the query in a hook:

```js
const setAuthentication = context => {
  const { accessToken, ...query } = context.params.query;

  context.params.authentication = {
    strategy: 'jwt',
    accessToken
  }
  // For subsequent requests, use the query without `accessToken`
  context.params = {
    ...context.params,
    query
  }
}
````

Then on the client you can do:

```js
const results = await app.service("messages").find({
  query: {
    ...query,
    accessToken
  }
});
```

Keep in mind that Socket.io connections will still maintain a connection for the user that originally authenticated (if any).

**Reply by palmtown** [reply]
Hello @daffl 

Thanks for your proposed solution.  Noted in regards to socket.io.  This should fulfill my requirements.

**Reply by palmtown** [reply]
Just an F.Y.I., this worked perfectly, thank you!
